core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -30,30 +32,36 @@
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
 import static io.trino.execution.scheduler.ScheduleResult.BlockedReason.WRITER_SCALING;
 import static io.trino.spi.StandardErrorCode.NO_NODES_AVAILABLE;
 import static io.trino.util.Failures.checkCondition;
+import static java.lang.Double.max;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public class ScaledWriterScheduler
         implements StageScheduler
 {
+    private static final double SCALE_WEIGHT_BASE = 100000.0;
```

Comment:
It's an arbitrary number to make the total weight smaller. But, it seems superfluous since there would be no need to make it smaller in respect to the max value of a double type. Let me remove it.